Explain cases to wait for SVCB (#101)
tfpauly/draft-happy-eyeballs-v3: draft-ietf-happy-happyeyeballs-v3.md

parameters and the SVCB ALPN set, as specified in {{Section 9.3 of
SVCB}}.

## Dropping or Pending Connection Attempts

Some situations related to handling SVCB responses can
require connection attempts to be dropped, or pended until
SVCB responses return.

{{Section 3.1 of SVCB}} describes client behavior for handling
resolution failures when responses are "cryptographically protected"
using DNSSEC {{?DNSSEC=RFC9364}} or encrypted DNS ({{?DOT=RFC7858}},
{{?DOH=RFC8484}}, or {{?DOQ=RFC9250}}, for example). If SVCB
resolution fails when using cryptographic protection, clients

> SHOULD abandon connection attempts to the altogether to avoid

Reviewer: ```suggestion
SHOULD abandon connection attempts altogether to avoid
```

downgrade attacks.

Use of cryptographic protection in DNS can influence other
parts of Happy Eyeballs connection establishment, as well.

Situations in which DNS is not protected allow for any records
to be blocked or modified, so security properties derived from
SVCB records are opportunistic only. However, when DNS is cryptographically
protected, clients can be stricter about relying on the properties
in SVCB records.

{{Section 5.1 of SVCB}} explains that clients "MUST NOT transmit any
information that could be altered by the SVCB response until it arrives",

> and specifically points about properties that affect the TLS ClientHello.

Reviewer: ```suggestion
and specifically mentions properties that affect the TLS ClientHello.
```

> This restriction specifically apply when a client's behavior will

Reviewer 2: ```suggestion
This restriction specifically applies when a client's behavior will
```

be altered by the SVCB response, which depends both on the client
implementation's ability to support a particular feature, as well
as the client implementation's willingness to rely on the SVCB

> response to enable a particular feature.

Reviewer 2: I'm not sure I really parse the last part of this sentence, what is it trying to say?

Based on this, clients in some scenarios MUST pend starting
a TLS handshake (either after TCP or as part of QUIC) until SVCB
responses have been received, even after the "Resolution Delay"
defined in {{resolution}} has been reached. Specifically, clients
MUST pend starting handshakes if *all* of the following are true:

1. DNS responses are cryptographically protected with DNSSEC or
encrypted DNS. (If unencrypted and unsigned DNS is used, SVCB
information is opportunistic; clients MAY wait for SVCB responses

> but do not need to.)

Reviewer 2: Trying to reduce parenthesis in this section:
```suggestion
encrypted DNS. Note that, if unencrypted and unsigned DNS is used, SVCB
information is opportunistic; clients MAY wait for SVCB responses
but do not need to.
```

> but do not need to.)

Reviewer: Trying to reduce parenthesis in this section:
```suggestion
encrypted DNS. Note that, if unencrypted and unsigned DNS is used, SVCB
information is opportunistic; clients MAY wait for SVCB responses
but do not need to.
```
